### PR TITLE
Updated Vert.x 4.4.3

### PR DESCRIPTION
**Pull request by ppatierno**
Trivial PR to update Vert.x 4.4.3 and aligning with the Strimzi bridge as well.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -132,8 +132,8 @@
         <!-- This is used to override CVE-2022-25857 and CVE-2022-41854 => should be removed once jackson-dataformat-yaml uses it out of the box -->
         <!-- https://github.com/strimzi/strimzi-kafka-operator/issues/7261 covers the removal of this override -->
         <snakeyaml.version>2.0</snakeyaml.version>
-        <vertx.version>4.3.8</vertx.version>
-        <vertx-junit5.version>4.3.8</vertx-junit5.version>
+        <vertx.version>4.4.3</vertx.version>
+        <vertx-junit5.version>4.4.3</vertx-junit5.version>
```
Comment (scholzj) [inline pom.xml:136]: Why not use 4.4.4?

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * Add support for _Kafka node pools_ according to [Strimzi Proposal #50](https://github.com/strimzi/proposals/blob/main/050-Kafka-Node-Pools.md)
 * Update OpenTelemetry 1.19.0
 * Fixed ordering of JVM performance options [#8579](https://github.com/strimzi/strimzi-kafka-operator/issues/8579)
+* Upgrade Vert.x to 4.4.3
```
Comment (scholzj) [inline CHANGELOG.md:12]: I probably wouldn't mention it here. It is just an internal implementation detail without any meaning to users.

Comment (ppatierno) [reply CHANGELOG.md:12]: I had doubt about it, I saw that we put this in some versions but not others :-)
Anyway I can remove, yes.

Comment (ppatierno) [reply CHANGELOG.md:12]: We have warnings which make the build failing because of some deprecations, I will take a look.

**Review by scholzj — APPROVED** [review]

**Pull request merged**